src/dotnet/commands/dotnet-build/ProjectGraphCollector.cs

```diff
@@ -44,7 +44,7 @@ private ProjectGraphNode TraverseProject(ProjectDescription project, IDictionary
                     LibraryDescription libraryDescription;
                     if (lookup.TryGetValue(dependency.Name, out libraryDescription))
                     {
-                        if (libraryDescription.Identity.Type.Equals(LibraryType.Project))
+                        if (libraryDescription.Resolved && libraryDescription.Identity.Type.Equals(LibraryType.Project))
```

Comment:
It is an error and it will be caught later. This was causing an InvalidCastException.

Before compiling, we do check if all libraries are resolved.